portlayer/exec/exec.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exec
+
+import (
+	"fmt"
+	"math/rand"
+	"net"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/vic/metadata"
+	"github.com/vmware/vic/pkg/trace"
+	"github.com/vmware/vic/pkg/vsphere/guest"
+	"github.com/vmware/vic/pkg/vsphere/session"
+	"github.com/vmware/vic/pkg/vsphere/spec"
+	"github.com/vmware/vic/pkg/vsphere/tasks"
+	"github.com/vmware/vic/pkg/vsphere/vm"
+	"golang.org/x/net/context"
+)
+
+const (
+	serialOverLANPort  = 2377
+	managementHostName = "management.localhost"
+)
+
+// Executor type for exec operations
+type Executor struct {
+	// Pointer to the session struct
+	session *session.Session
+}
+
+// ExecutorCreateConfig defines the parameters for Create call
+type ExecutorCreateConfig struct {
+	Metadata metadata.ExecutorConfig
+
+	ParentImageID  string
+	ImageStoreName string
+	VCHName        string
+}
+
+// ExecutorStartConfig defines the parameters for Start call
+type ExecutorStartConfig struct {
+	ID string
+}
+
+// NewExecutor returns a new Executor
+func NewExecutor(ctx context.Context, session *session.Session) (*Executor, error) {
+	e := &Executor{
+		session: session,
+	}
+
+	return e, nil
+}
+
+// Create creates a container vm with the given params
+func (e *Executor) Create(ctx context.Context, config *ExecutorCreateConfig) error {
+	defer trace.End(trace.Begin("Create"))
+
+	// Convert the management hostname to IP
+	ips, err := net.LookupIP(managementHostName)
```

Comment:
torn about whether this should be done at initialization - any idea what the latency is on a hostname lookup from /etc/hosts?